core/main.js

```diff
@@ -0,0 +1,291 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The entrypoint for blockly_compressed.js.  Provides
+ *     various backwards-compatibility hacks.  Not used when loading
+ *     in uncompiled (uncompressed) mode via bootstrap.js.
+ */
+'use strict';
+
+goog.module('Blockly.main');
+
+const Blockly = goog.require('Blockly');
+const ContextMenu = goog.require('Blockly.ContextMenu');
+const Events = goog.require('Blockly.Events');
+const Msg = goog.require('Blockly.Msg');
+const Tooltip = goog.require('Blockly.Tooltip');
+const WidgetDiv = goog.require('Blockly.WidgetDiv');
+const colour = goog.require('Blockly.utils.colour');
+const common = goog.require('Blockly.common');
+const deprecation = goog.require('Blockly.utils.deprecation');
+const dialog = goog.require('Blockly.dialog');
+const eventUtils = goog.require('Blockly.Events.utils');
+
+/*
+ * Aliased functions and properties that used to be on the Blockly namespace.
+ * Everything in this section is deprecated. Both external and internal code
+ * should avoid using these functions and use the designated replacements.
+ * Everything in this section will be removed in a future version of Blockly.
+ */
+
+// Add accessors for properties on Blockly that have now been deprecated.
+Object.defineProperties(Blockly, {
+  /**
+   * Wrapper to window.alert() that app developers may override to
+   * provide alternatives to the modal browser window.
+   * @name Blockly.alert
+   * @type {!function(string, function()=)}
+   * @deprecated Use Blockly.dialog.alert / .setAlert() instead.
+   *     (December 2021)
+   * @suppress {checkTypes}
+   */
+  alert: {
+    set: function(newAlert) {
+      deprecation.warn('Blockly.alert', 'December 2021', 'December 2022');
+      dialog.setAlert(newAlert);
+    },
+    get: function() {
+      deprecation.warn(
+          'Blockly.alert', 'December 2021', 'December 2022',
+          'Blockly.dialog.alert()');
+      return dialog.alert;
+    },
+  },
+  /**
+   * Wrapper to window.confirm() that app developers may override to
+   * provide alternatives to the modal browser window.
+   * @name Blockly.confirm
+   * @type {!function(string, function()=)}
+   * @deprecated Use Blockly.dialog.confirm / .setConfirm() instead.
+   *     (December 2021)
+   * @suppress {checkTypes}
+   */
+  confirm: {
+    set: function(newConfirm) {
+      deprecation.warn('Blockly.confirm', 'December 2021', 'December 2022');
```

Comment:
When we add deprecation warnings to `mainWorkspace` and friends, I think we should change this to be `'September 2022'`